library/alloc/src/boxed.rs

```diff
@@ -518,9 +756,9 @@ impl<T: ?Sized> Box<T> {
     /// ```
     #[stable(feature = "box_leak", since = "1.26.0")]
     #[inline]
-    pub fn leak<'a>(b: Box<T>) -> &'a mut T
+    pub fn leak<'a>(b: Self) -> &'a mut T
     where
-        T: 'a, // Technically not needed, but kept to be explicit.
+        A: 'a,
```

Comment:
I added a test, which ensures, that the allocator must not be moved while the leaked data is in use.